mitmproxy/contrib/tnetstring.py

```diff
@@ -158,6 +158,8 @@ def _rdumpq(q, size, value):
         write(b':')
         write(span)
         return size + 2 + len(span) + lvalue
+    elif isinstance(value, str):
+        return _rdumpq(q, size, value.encode())
```

Comment:
The underlying issue seems to be that we have unicode/str in `.get_state()`. I feel we should rather fix that rather than hoping that we can encode here. Which part triggers this?